### PR TITLE
Volume normalization

### PR DESCRIPTION
**Pull request by herrernst**
See #15

### CODE REVIEW

**Comment by ComlOnline** [comment]
@herrernst You're conflict in main is due to the zeroconf-port option addition. 
And in player its due to all the commits since we forked from plietar. 
You can see a list here:
https://github.com/librespot-org/librespot/commits/master/src/player.rs

**Comment by herrernst** [comment]
@ComlOnline Feel free to resolve them ;)

**Comment by ComlOnline** [comment]
I will do but it will take some time as github wont let me do it online.

**Comment by ComlOnline** [comment]
That should have done it, compiling now to check.

File: core/src/config.rs [hunk 1]
```diff
@@ -105,6 +105,8 @@ pub struct PlayerConfig {
     pub bitrate: Bitrate,
     pub onstart: Option<String>,
     pub onstop: Option<String>,
+    pub normalization: bool,
+    pub normalization_pre_gain: f32,
```
Comment (plietar) [inline core/src/config.rs]: Use `pregain` as a single word, here and elsewhere.

File: src/player.rs [hunk 2]
```diff
@@ -257,13 +266,22 @@ impl PlayerInternal {
...
+    fn handle_packet(&mut self, packet: Option<VorbisPacket>, normalization_factor: f32) {
         match packet {
             Some(mut packet) => {
                 if let Some(ref editor) = self.audio_filter {
                     editor.modify_stream(&mut packet.data_mut())
                 };
+              
+                if self.config.normalization {
 
+                    if normalization_factor != 1.0 {
```
Comment (plietar) [inline src/player.rs]: Combine these two ifs

File: src/player.rs [hunk 3]
```diff
@@ -430,7 +450,59 @@ impl PlayerInternal {
...
+        let mut decrypted_file = AudioDecrypt::new(key, encrypted_file);
+
+        let mut normalization_factor: f32 = 1.0;
+
+        if self.config.normalization {
+            //buffer for float bytes
+            let mut track_gain_float_bytes = [0; 4];
+
+            decrypted_file.seek(SeekFrom::Start(144)).unwrap(); // 4 bytes as LE float
+            decrypted_file.read(&mut track_gain_float_bytes).unwrap();
```
Comment (plietar) [inline src/player.rs]: You should use [`read_f32::<LittleEndian>`](https://docs.rs/byteorder/1.2.1/byteorder/trait.ReadBytesExt.html#method.read_f32)

File: src/player.rs [hunk 4]
```diff
@@ -430,7 +450,59 @@ impl PlayerInternal {
...
 
         let encrypted_file = AudioFile::open(&self.session, file_id).wait().unwrap();
-        let audio_file = Subfile::new(AudioDecrypt::new(key, encrypted_file), 0xa7);
+
+        let mut decrypted_file = AudioDecrypt::new(key, encrypted_file);
+
+        let mut normalization_factor: f32 = 1.0;
+
+        if self.config.normalization {
+            //buffer for float bytes
```
Comment (plietar) [inline src/player.rs]: Break this up into a function which does the parsing and returns a struct with 4 f32s. You can `#[derive(Debug)]` and use a single debug statement.

**Comment by sashahilton00** [comment]
@herrernst pinging in case forgotten.

**Pull request closed without merging**